adapter/src/dbt/include/fal/macros/teleport_snowflake.sql

```diff
@@ -0,0 +1,21 @@
+{% macro snowflake__copy_from(relation, location, columns) -%}
+    CREATE OR REPLACE TABLE {{ relation }} USING template (
+      SELECT array_agg(object_construct(*))
+	  	FROM table(
+  			infer_schema(
+	  			LOCATION=>'{{ location }}',
+		  		file_format=>'falparquet'
+			  )
+  		)
+    );
+    COPY INTO {{ relation }} FROM (SELECT {{ columns }} FROM {{ location }})
```

Comment:
I think we could maybe break this into 2 macros?

1. Macro with `CREATE OR REPLEACE TABLE...`
2. Fetch the newly created table's column names
  a. could be `DESCRIBE {{ relation }}`
  b. could be a dbt method `adatper.fetch_columns` or maybe reloading the relation info will bring the columns?
  c. could be `SELECT column_name FROM {{ relation.database }}.information_schema.columns WHERE table_name ilike '{{ relation.identifier }}';`
3. Macro with `COPY INTO` that get's the columns passed
  a. A comment: you could pass the `columns` list, and do a for-loop inside Jinja